File: backend/Models/Question.cs
using System.ComponentModel.DataAnnotations;

namespace prid_2324_a02.Models;

public class Question
{
	[Key]
	public int Id { get; set; }
	public int Order { get; set; }
	public string Body { get; set; } = null!;

	public Quizz Quizz { get; set; } = null!;
	public int QuizzId { get; set; }

	public virtual ICollection<Solution> Solutions { get; set; } = new HashSet<Solution>();
	public virtual ICollection<Answer> Answers { get; set; } = new HashSet<Answer>();

}

File: frontend/src/app/models/question.ts
import { Answer } from "./answer";
import { Attempt } from "./attempt";
import { Quiz } from "./quiz";
import { Solution } from "./solution";

export class Question {
	id?: number;
	order?: number;
	body?: string;
	quiz?: Quiz;

	solutions: Solution[] = [];

	previous: number = 0;
	next: number = 0;
	answer?: Answer | undefined = undefined;
	attempt?: Attempt | undefined = undefined;
}